### ISSUE TITLE
Function parameter complete error on centos7, vim 7.4.649

### ISSUE THREAD

**Issue by aloneneutrino** [wsdjeg/SpaceVim#3709]
## Expected behavior, english is required
CentOS 7 , vim 7.4.649
Input ```printf(``` , and some error happened.
## The reproduce ways from Vim starting (Required!)
Input ```printf(```  or other function when editing C file.
## Debug info

Please press <kbd>SPC h I</kbd>, debug info will be put into clipboard, then paste all content below.
It runs on a remote server, so there is no access to clipboard.
## Screenshots

If you have any screenshots for this issue, please upload here. BTW you can use https://asciinema.org/ for recording video in terminal.
![image](https://user-images.githubusercontent.com/35491491/90332354-a152f380-dfee-11ea-9ab1-f1b7ff1fad2d.png)

There are other errors happen when using vim 7.4.649, so I installed vim8 and it works well.
Should we update the document to "recommend to use vim8" ?

**Comment by wsdjeg** [comment]
@aloneneutrino sorry, This seems a bug of complete_parameter.vim, the `v:complete_item` is new feature. BTW, the tested version of SpaceVim is `7.4.1689`. `7.4.649` is too old. 

To fix this issue, we need to add version checker in SpaceVim, or implement this feature in other way. Before doing this, may I ask, is `7.4.649` the default version of vim in centos 7?

**Comment by Shougo** [comment]
I think SpaceVim should not support Vim 7.4...
It is too old in these days.

**Comment by wsdjeg** [comment]
I know, But centos 7 is not too old. I use this os too. And I have many vps which use this os, I also do not want to build Vim in each of them.

**Comment by Shougo** [comment]
Hm.  If so SpaceVim need to check the variable exists

**Comment by wsdjeg** [comment]
yeah, too many works need to be done. I am try to fix lint in #3727

**Comment by aloneneutrino** [comment]
I think ``7.4.629`` is the default version in the ``os`` repo for CentOS 7.
```
# yum list | grep vim
Repository epel is listed more than once in the configuration
vim-common.x86_64                         2:7.4.629-6.el7              @os
vim-enhanced.x86_64                       2:7.4.629-6.el7              @os
vim-filesystem.x86_64                     2:7.4.160-4.el7              @anaconda
vim-minimal.x86_64                        2:7.4.160-4.el7              @anaconda
beakerlib-vim-syntax.noarch               1.18-12.el7                  epel
boxes-vim.noarch                          1.3-4.el7                    epel
fluxbox-vim-syntax.noarch                 1.3.7-1.el7                  epel
geany-plugins-vimode.x86_64               1.36-1.el7                   epel
neovim.x86_64                             0.3.0-2.el7                  epel
protobuf-vim.x86_64                       2.5.0-8.el7                  os
python2-neovim.noarch                     0.3.2-1.el7                  epel
python36-neovim.noarch                    0.3.2-1.el7                  epel
vim-X11.x86_64                            2:7.4.629-6.el7              os
vim-filesystem.x86_64                     2:7.4.629-6.el7              os
vim-fugitive.noarch                       2.5-1.el7                    epel
vim-go.x86_64                             1.8-3.el7                    epel
vim-gtk-syntax.noarch                     20130716-1.el7               epel
vim-halibut.noarch                        1.2-1.el7                    epel
vim-jellybeans.noarch                     1.7-1.el7                    epel
vim-minimal.x86_64                        2:7.4.629-6.el7              os
vim-toml.noarch                           0-0.2.20180615git85ba827.el7 epel
vim-trailing-whitespace.noarch            1.0-1.20191209git6b7cdec.el7 epel
vim-vimoutliner.noarch                    0.4.0-8.el7                  epel
xtuple-csvimp.x86_64                      0.5.4-8.el7                  epel
xtuple-csvimp-devel.x86_64                0.5.4-8.el7                  epel
```
![image](https://user-images.githubusercontent.com/35491491/91583943-7e2c2a80-e984-11ea-8aaf-164320cd5dfa.png)
But 7.4 is too old indeed, maybe the better solution is to install vim8 for centos7.
